lambda/translprim.ml

```diff
@@ -658,6 +680,75 @@ let lambda_of_loc kind sloc =
     let scope_name = Debuginfo.Scoped_location.string_of_scoped_location sloc in
     Lconst (Const_immstring scope_name)
 
+let atomic_arity (op : atomic_op) (kind : atomic_kind) =
+  let arity_of_op = match op with
+    | Load -> 1
+    | Exchange -> 2
+    | Cas -> 3
+    | Fetch_add -> 2
+  in
+  let extra_kind_arity = match kind with
+    | Ref | Loc -> 0
+    | Record -> 1
+  in
+  arity_of_op + extra_kind_arity
+
+let lambda_of_atomic prim_name loc op kind args =
+  if List.length args <> atomic_arity op kind then
+    raise(Error(to_location loc,
+                Wrong_arity_builtin_primitive prim_name));
+  let split = function
+    | [] ->
+        (* split is only called when [arity >= 1] *)
+        assert false
+    | first :: rest -> first, rest
+  in
+  let prim = match (op : atomic_op) with
+    | Load -> Patomic_load
+    | Exchange -> Pccall prim_atomic_exchange
+    | Cas -> Pccall prim_atomic_cas
+    | Fetch_add -> Pccall prim_atomic_fetch_add
+  in
+  match (kind : atomic_kind) with
+  | Record ->
+      (* the primitive application
```

Comment:
Unsure why this was marked as resolved. Do you disagree?

Reply:
My memories from our apply-comments-and-rebase session is as follows: (1) yes, we were not fully enthusiastic about the proposal which breaks regularity and reads a bit heavily, but (2) rather than just ignoring it, we decided to rewrite the comments so that they are easier to understand, and in particular @clef-men suggested writing the examples as pseudo-AST rather than pseudo-code, and I think they read nicer, and finally (3) I assumed that the significant changes to the comment had made the point moot, in the sense that the structure would now be different anyway, and we wouldn't have to be fully explicit on (1) (do we really disagree with Olivier here?) when it's maybe not a key part of the PR discussion, so I decided to consider the comment morally "outdated" and make it go away. In retrospect (looking at the old and new comments) the changes are not as invasive as I thought they were, and (3) is not fully convincing, but I propose to keep it as-is and look the other way, if that's fine with you. (Otherwise please do insist and I think we'll come back to add "For example," in each case.)

Reply:
OK, I won’t press it.